e2e/aws_test.go

```diff
@@ -47,6 +47,9 @@ var _ = Describe("Cluster API AWS MachineSet", Ordered, func() {
 	})
 
 	AfterEach(func() {
+		if platform != configv1.AWSPlatformType {
+			Skip("Skipping AWS E2E tests")
+		}
```

Comment:
Why is this not covered by the `Skip` in the `BeforeAll`? I think it's because AfterEach and AfterAll always run even when a test is skipped, we should have a comment in the code to make it obvious